dart/lib/src/hub.dart

```diff
@@ -0,0 +1,379 @@
+import 'dart:async';
+import 'dart:collection';
+
+import 'client.dart';
+import 'protocol.dart';
+import 'scope.dart';
+import 'sentry_options.dart';
+
+typedef ScopeCallback = void Function(Scope);
+
+/// SDK API contract which combines a client and scope management
+class Hub implements HubInterface {
```

Comment:
for the interfaces, is it common to use Name`Interface` in Dart? if theres not really a default pattern, we could do like Java, IHub

Reply:
Actually I wonder if we need one. It helped me to list the methods but IMO the Dart way would be to not have an interface at all here. The NoOpHub could implements Hub directly ?

Reply:
we use the NoOp pattern because Java, its not null-safety, I believe here we'd need to.
the reason for using interfaces is to be able to test/mock them.

Also, some classes could be overwritten by the customer, if they prefer to have their own implementation, not really needed for Hub and Client I guess, but its common to see for the Transport (which we still dont have for example)

Reply:
For mocking :
```
class MockHub extends Mock implements Hub{}
```

Reply:
if in Dart a class could `implements` any other class, maybe interfaces are not really necessary, good point

Reply:
I finally removed it ! unless we don't want expose some Hub methods, it makes no difference.

Reply:
then every non exposed method should be private